python/activator/middleware_interface.py

```diff
@@ -670,14 +680,25 @@ def _export_collections(self, collection):
 
         # Store collection chains after all children guaranteed to exist
         chains = {}
+        removed = set()
         for child in src.queryCollections(collection, flattenChains=True, includeChains=True):
+            # VALIDITY-HACK: do not transfer calibration collections; we'll make our own
+            if src.getCollectionType(child) == CollectionType.CALIBRATION:
+                removed.add(child)
+                continue
             if src.getCollectionType(child) == CollectionType.CHAINED:
                 chains[child] = src.getCollectionChain(child)
             dest.registerCollection(child,
                                     src.getCollectionType(child),
                                     src.getCollectionDocumentation(child))
         for chain, children in chains.items():
-            dest.setCollectionChain(chain, children)
+            # VALIDITY-HACK: fix up chains after removing calibration
+            # collections and including local-only collections.
+            present = list(dest.getCollectionChain(chain)) + list(children)
+            for missing in removed:
+                while missing in present:
+                    present.remove(missing)
+            dest.setCollectionChain(chain, present)
```

Comment:
The intent of the original code (before I hacked it up) was to perfectly reproduce the appropriate subset of the central repo locally. Is there a way to do that?

Deliberately transforming everything into a flat collection sounds like it should work (including with overday updates to the central repo), but I'm worried there might be some hidden assumptions built in.

Reply:
To just fix the duplication you'd need to pass `flattenChains=False` and recurse whenever you see a CHAINED collection.

Reply:
Wait -- are you sure the original code is wrong? I'm pretty sure the first loop visits each collection (chain or leaf) exactly once, and registers only that collection, and not any children, in `dest`.

Reply:
Oh, you're right.  Original code has the right behavior.  Sorry for the noise.

Reply:
After some more thought, I won't try to make the final state into a flat collection -- it will make this change harder to roll back, and I definitely don't want it to be the permanent solution. 😱